commands/deploy.go

```diff
@@ -326,3 +324,26 @@ func deriveFprocess(function stack.Function) (string, error) {
 func languageExistsNotDockerfile(language string) bool {
 	return len(language) > 0 && strings.ToLower(language) != "dockerfile"
 }
+
+func compileSecrets(yamlSecrets []string, cmdLineSecrets []string) []string {
+	secretsMap := make(map[string]bool)
```

Comment:
I was thinking something like what was done for labels/env-vars? Can this be used? https://github.com/ericstoekl/faas-cli/blob/e5d279cfdde669fbc6a51ad42fe003a8f2d02ed4/commands/deploy.go#L264